### PR TITLE
Feature: Associations

### PR DESCRIPTION
**Pull request by AdornChoga**
In this pull request, I did the following:
- Created  `Classroom` , `Book` and `Rental` classes
- Created the has-many/belongs-to relationship between `Classroom` and `Student`
- Created the `many-to-many` (also `has-many-through`) relationship 
  between `Person` and `Book` using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Amrhub) [inline book.rb:9]: - [ ] We forgot to add a many-to-many relationship between `Book` and `Person` Which should be through `Rental` so `Rental` has a 1-to-many relationship with `Book` here we need to add a one-to-many relationship by defining a function for that
```ruby
def add_rental(person,date)
  Rental.new(person,date,self)
 end
 ```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+
```
Comment (Amrhub) [inline rental.rb]: - [ ] Let's also make sure we add attr_accessor for `:book, :person` you will need it when displaying on console later on 😇 so you can access person/book information through your rental

File: person.rb [hunk 3]
```diff
@@ -1,42 +1,15 @@
...
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', _parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
+    @rentals = []
```
Comment (Amrhub) [inline person.rb:12]: - [ ] As mentioned above in `book.rb` let's kindly make sure we also have one-to-many relationship between `Person` and `Rental` by defining a method here as well with the suitable changes 👍🏻

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  def initialize(person, book, date)
+    @date = date
+    @person = person
+    @book = book
+
+    person.rentals << self
+    book.rentals << self
```
Comment (Amrhub) [inline rental.rb:10]: Great job implementing `belongs_to` that's why I assumed in `book.rb` that we forgot to add `has_many` relationship ☺️

**Review by Amrhub — CHANGES_REQUESTED** [review]
Hi @AdornChoga,  👋🏻 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!
These changes aim to add some awesomeness to your awesome project and code skills.

## To Highlight ✔️ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 


## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 4 above)
Comment (AdornChoga) [reply rental.rb:10]: Thank you very much for your feedback @Amrhub. Based on the last comment, does that mean [this change](https://github.com/AdornChoga/oop-school-library/pull/3#discussion_r843127254) is the only change I need to make now?

Comment (Amrhub) [reply rental.rb:10]: You are most welcome @AdornChoga, any comment with the checkbox at the start is the changes we need to make actually this comment you linked is supposed to be OPTIONAL I changed it, so kindly also make association `has_many` between book to rental and between person to rental

Comment (AdornChoga) [reply rental.rb:10]: Thank you for the clarification @Amrhub. I'm doing that right away.

**Review by hiromataba — APPROVED** [review]
# STATUS: APPROVED :tada: :tada: :tada: 
Hi @AdornChoga         :wave: :wave: 

appr

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**